### PR TITLE
Prompt user to reauthenticate if necessary during long running jobs

### PR DESCRIPTION
**Pull request by jacoblurye**
This PR introduces the `retry_with_reauth` decorator for API requests. If a request fails with status code 403, the decorator will prompt the user to provide a fresh ID token then retry the request with that new token.

`retry_with_reauth` reads users' ID tokens directly from their clipboard, because ID tokens are often longer than the 1024 byte limit on the stdin buffer for some terminals (e.g., MacOS's).

### CODE REVIEW

File: cli/api.py [hunk 1]
```diff
@@ -53,8 +63,53 @@ def check_auth(id_token: str) -> Optional[str]:
...
+                click.prompt(
+                    (
+                        "\nCIDC reauthentication required. Please copy a fresh identity token from the Portal "
+                        f"to your clipboard here:\n\n\t{TOKEN_URL}\n\n"
+                        "Then, press 'enter' to paste your copied token below"
+                    ),
+                    default="enter",
+                    show_default=False,
+                )
+                id_token = _read_clipboard()
```
Comment (curlup) [inline cli/api.py]: exception handling here? what if there's no display/clipboard for Tk to operate on?

File: cli/api.py [hunk 2]
```diff
@@ -53,8 +63,53 @@ def check_auth(id_token: str) -> Optional[str]:
...
+            # If the error isn't auth-related, break out of the retry loop.
+            if res.status_code != 403:
+                break
+
+            # Prompt the user for a new ID token.
+            while True:
+                click.prompt(
+                    (
+                        "\nCIDC reauthentication required. Please copy a fresh identity token from the Portal "
+                        f"to your clipboard here:\n\n\t{TOKEN_URL}\n\n"
```
Comment (curlup) [inline cli/api.py]: nit: Isn't "... from the Portal to clipboard here:" a little confusing?

File: cli/api.py [hunk 3]
```diff
@@ -170,10 +226,12 @@ def poll_upload_merge_status(job_id: int) -> MergeStatus:
     """Check the merge status of an upload job"""
     url = _url(f"/ingestion/poll_upload_merge_status")
     params = dict(id=job_id)
-    response = requests.get(url, params=params, headers=_with_auth())
 
-    if response.status_code != 200:
-        raise ApiError(_error_message(response))
+    @retry_with_reauth
+    def make_request():
+        return requests.get(url, params=params, headers=_with_auth())
```
Comment (curlup) [inline cli/api.py]: aren't you basically wrapping all `requests.xxx`s? Is there a way maybe to DRY? I would suggest monkeypatching `requests`, though I know it's a little controversial.

Comment (jacoblurye) [reply cli/api.py]: Hmm, maybe!

File: cli/api.py [hunk 4]
```diff
@@ -112,6 +113,18 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+class RequestsWithReauth:
```
Comment (curlup) [inline cli/api.py]: if it's a service (private?) class, wouldn't it be right to name it starting from underscore or something?

**Pull request merged**